<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
</head>

<body>
  <script>
    var p = Promise.resolve()
      .then(() => {
        // 使 .then() 返回一个 rejected promise
        throw new Error('Oh no!');
      })
      .catch(error => {
        console.error('onRejected function called: ' + error.message);
      })
      .then(() => {
        console.log("I am always called even if the prior then's promise rejects");
      });
    
  </script>
</body>

</html>